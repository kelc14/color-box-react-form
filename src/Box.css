.Box {
  position: relative;
}
.Box-btn {
  position: absolute;
  right: 0;
  bottom: 0;
}
