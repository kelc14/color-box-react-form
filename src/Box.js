import React, { useState } from "react";
import "./Box.css";

const Box = ({ backgroundColor, width, height, removeBox, id }) => {
  const handleRemove = (e) => {
    let boxId = e.target.dataset.box;
    removeBox(boxId);
  };
  /** Render
   * - box item with:
   * backgroundColor,
   * width
   * and height
   *
   * and a button to remove the box
   * */
  return (
    <>
      <div
        className="Box"
        id={id}
        style={{
          backgroundColor,
          width: `${width}px`,
          height: `${height}px`,
        }}
      >
        <button onClick={handleRemove} data-box={id} className="Box-btn">
          X
        </button>
      </div>
    </>
  );
};

export default Box;
