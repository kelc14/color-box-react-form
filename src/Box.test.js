import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

// smoke test
it("renders without crashing", function () {
  render(<Box key={1} id={1} backgroundColor="red" width={200} height={200} />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <Box key={1} id={1} backgroundColor="red" width={200} height={200} />
  );
  expect(asFragment()).toMatchSnapshot();
});
