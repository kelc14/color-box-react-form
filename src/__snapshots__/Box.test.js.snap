// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    id="1"
    style="background-color: red; width: 200px; height: 200px;"
  >
    <button
      class="Box-btn"
      data-box="1"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
